e2e_playwright/st_plotly_chart_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pytest
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+# Only do chromium as this can create a lot of screenshots
+# there should be no differences between chrome and safari and firefox
+@pytest.mark.only_browser("chromium")
+def test_plotly_has_consistent_visuals(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    expect(themed_app.locator(".element-container")).to_have_count(16)
+    assert_snapshot(
```

Comment:
Can we rewrite these to be something like

```python3
snapshot_names = [
    "st_plotly_chart-none-theme",
    "st_plotly_chart-streamlit-theme-use-container-width",
    # ...
]
for i, name in enumerate(snapshot_names):
    assert_snapshot(
        themed_app.locator(".element-container .stPlotlyChart").nth(i),
        name=name,
    )
```

since the only difference between all of these `assert_snapshot` calls is the index of the chart and the snapshot name?

Reply:
Good point. Makes it a lot cleaner. thanks.